src/DataWrangling/ECCO/ECCO_metadata.jl

```diff
@@ -217,30 +219,37 @@ function download_dataset(metadata::ECCOMetadata; url = urls(metadata))
     username = get(ENV, "ECCO_USERNAME", nothing)
     password = get(ENV, "ECCO_PASSWORD", nothing)
     dir = metadata.dir
-
-    @distribute for metadatum in metadata # Distribute the download among ranks if MPI is initialized
-
-        fileurl  = metadata_url(url, metadatum) 
-        filepath = metadata_path(metadatum)
-
-        if !isfile(filepath)
-            instructions_msg = "\n See ClimaOcean.jl/src/ECCO/README.md for instructions."
-            if isnothing(username)
-                msg = "Could not find the ECCO_PASSWORD environment variable. \
-                       See ClimaOcean.jl/src/ECCO/README.md for instructions on obtaining \
-                       and setting your ECCO_USERNAME and ECCO_PASSWORD." * instructions_msg
-                throw(ArgumentError(msg))
-            elseif isnothing(password)
-                msg = "Could not find the ECCO_PASSWORD environment variable. \
-                       See ClimaOcean.jl/src/ECCO/README.md for instructions on obtaining \
-                       and setting your ECCO_USERNAME and ECCO_PASSWORD." * instructions_msg
-                throw(ArgumentError(msg))
+    
+    # Create a temporary directory to store the .netrc file
+    # The directory will be deleted after the download is complete
+    mktempdir(dir) do tmp
+
+        # Write down the username and password in a .netrc file
+        downloader = netrc_downloader(username, password, "ecco.jpl.nasa.gov", tmp)
+
+        @distribute for metadatum in metadata # Distribute the download among ranks if MPI is initialized
```

Comment:
Having multiple processes download the same file to the same location can lead to race conditions. It is also the best of the etiquettes: if you run with 1000 MPI processes, this will send 1000 requests to the ECCO servers at the very same time, which would lead to high load on their end.

Reply:
the `@distribute` macro distributed the download over the ranks, so in theory we should never have different ranks downloading the same file. The second comment is right though, if we have 1000 ranks and 1000 files to download we will have 1000 requests at the same time to the ECCO servers (different files though). Will this create problems?

Reply:
> the @distribute macro distributes the download over the ranks, so in theory we should never have different ranks downloading the same file.

Got it. Thanks 

> The second comment is right though, if we have 1000 ranks and 1000 files to download we will have 1000 requests at the same time to the ECCO servers (different files though). Will this create problems?

I don't think it's necessarily a problem, but it's something to be mindful about (mostly as a pattern to not abuse). If one is running on 1000s processes, it is probably best to pre-download the data

Reply:
It's also a hack, why would we distributed among MPI processes? It might be better to instead distribute with tasks, but only from rank 0. That way the download is distributed and fast, but we limit the number of requests to something reasonable.